### PR TITLE
Add v0.14 release candidates docs and cleanup

### PR DESCRIPTION
**Pull request by YosuaMichael**
- Add v0.14 release candidates docs in versions.html
- Cleanup the docs to use v0.14.0

### CODE REVIEW

**Review by svekars — APPROVED** [review]
LGTM! But @NicolasHug should take a look too

File: versions.html [hunk 1]
```diff
@@ -20,6 +20,9 @@ <h1> PyTorch Documentation </h1>
         <li class="toctree-l1">
           <a class="reference internal" href="main/">main (unstable)</a>
         </li>
+        <li class="toctree-l1">
+          <a class="reference internal" href="0.14/">v0.14 (release candidate)</a>
```
Comment (NicolasHug) [inline versions.html:24]: We'll need to remember to update that when the release is out for good

**Review by NicolasHug — APPROVED** [review]
Thanks Yosua, LGTM although I would suggest to rename `0.14.0` into simply `0.14`, as we usually don't mention the "bugfix  part" in the docs

File: versions.html (hunk 1 above)
Comment (YosuaMichael) [reply versions.html:24]: got it, thanks @NicolasHug , will update this

Comment (NicolasHug) [reply versions.html:24]: Sorry I just meant the `.0` part i.e. minor version

Comment (YosuaMichael) [reply versions.html:24]: Ah okay, I have fixed it then

**Pull request merged**